### PR TITLE
docs: add known limitations docs

### LINKED ISSUE #290: Sync of k8 secrets not happening
**What steps did you take and what happened:**
I have set up a SecretProviderClass that connects to my Azure Key Vault which works correctly as I can mount the volume and get the secrets out (as seen in the tutorial). I have added secretObjects to sync with th k8 secrets but the secret is not appearing. 
I'm trying to create a simple Opaque secret. Is the secret not created if it does not exist?

**What did you expect to happen:**
That a secret would be created and and synced.

**Which provider are you using:**
I'm using Azure Key Vault

**Environment:**

- Secrets Store CSI Driver version: (use the image tag): us.gcr.io/k8s-artifacts-prod/csi-secrets-store/driver:v0.0.12
- Kubernetes version: (use `kubectl version`): v1.17.7

### PR DESCRIPTION
**Pull request by aramase**
**What this PR does / why we need it**:
- Adds known limitations docs

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
fixes #290, fixes https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/231

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/README.limitations.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Known Limitations
+
+This document highlights the current limitations when using secrets-store-csi-driver.
+
+## Mounted content and Kubernetes Secret not updated after secret is updated in external secrets-store
+
+When the secret/key is updated in external secrets store after the inital pod deployment, the updated secret is not automatically reflected in the pod mount or the Kubernetes secret. 
+
+This feature is planned for release (`v0.0.15`). See [design doc](https://docs.google.com/document/d/1RGT0vmeUnN71n_u5fZKsSCa2YQpGw99rfGN9RlFMgHs/edit?usp=sharing) for more details.
```
Comment (ritazh) [inline docs/README.limitations.md:9]: Can we say `planned for release v0.0.15+`

File: docs/README.limitations.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+### How to fetch the latest content with the current releases?
```
Comment (ritazh) [inline docs/README.limitations.md:11]: Can we say `fetch the latest content with release v0.0.14 and earlier`

Comment (aramase) [reply docs/README.limitations.md]: Updated!

**Pull request merged**